site/app/controllers/student/SubmissionController.php

```diff
@@ -486,7 +486,24 @@ public function ajaxBulkUpload($gradeable_id) {
                     "is_qr"     => false
                 ];
 
-                $bulk_upload_job  = "/var/local/submitty/daemon_job_queue/bulk_upload_" . $uploaded_file["name"][$i] . ".json";
+                $bulk_upload_job  = FileUtils::joinPaths("/var/local/submitty/daemon_job_queue/bulk_upload_", $uploaded_file["name"][$i], ".json");
+
+                // exec() and similar functions are disabled by security policy,
+                // so we are using a python script via CGI to validate whether file is divisible by num_page or not.
+                $pdf_full_path = FileUtils::joinPaths($pdf_path, $job_data["timestamp"], $job_data["filename"]);
+                $ch = curl_init();
+                curl_setopt($ch, CURLOPT_URL, $this->core->getConfig()->getCgiUrl() . "pdf_page_check.cgi?pdf_path={$pdf_full_path}&num_page={$num_pages}&file_name={$job_data['filename']}");
```

Comment:
could we use `rawurlencode` for filename and pdf_full_path? maybe even for num_pages, URL encoding is good practice to handle unsafe characters.

Reply:
I think there are some example use of `rawurlencode` in our codebase. Feel free to use them as references.

Reply:
Sure.

Reply:
Done.